### PR TITLE
8343124: Tests fails with java.lang.IllegalAccessException: class com.sun.javatest.regtest.agent.MainWrapper cannot access

### PR DESCRIPTION
**Pull request by prsadhuk**
Couple of printing tests dont have "public" modifier so CI testing fails to run this test citing IllegalAccessException.
Fix is made to made the test class public.

Additionally, modified PrintDlgPageable.java to use PFJ and made StreamPrintingOrientation.java automated.

### CODE REVIEW

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 1]
```diff
@@ -24,95 +24,110 @@
...
+import javax.print.attribute.Attribute;
+import javax.print.PrintService;
+import javax.print.StreamPrintServiceFactory;
+import javax.print.attribute.standard.Copies;
+import javax.print.attribute.standard.JobName;
+import javax.print.attribute.standard.OrientationRequested;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+
+public class StreamPrintingOrientation implements Printable {
+    public StreamPrintingOrientation() {
```
Comment (TejeshR13) [inline test/jdk/javax/print/StreamPrintingOrientation.java:52]: Can remove this constructor.

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 2]
```diff
@@ -24,95 +24,110 @@
...
+        PrintService service = null;
 
-class StreamPrintingOrientation implements Printable {
-        /**
-         * Constructor
-         */
-         public StreamPrintingOrientation() {
-                super();
+        FileOutputStream fos = null;
+        File fl = null, fp = null;
```
Comment (TejeshR13) [inline test/jdk/javax/print/StreamPrintingOrientation.java:63]: `null` initialization is redundant here.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 3]
```diff
@@ -26,104 +26,121 @@
...
-        /**
-         * Starts the application.
-         */
-        public static void main(java.lang.String[] args) {
-            if (args.length < 1) {
-                System.out.println("usage: java PrintDlgPageable { 0 | 2}");
-                return;
-            }
-            arg = Integer.parseInt(args[0]);
-                PrintDlgPageable pd = new PrintDlgPageable();
```
Comment (TejeshR13) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:54]: `pd` unused.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 4]
```diff
@@ -26,104 +26,121 @@
...
+            }
+            catch (PrinterException pe) {
+                pe.printStackTrace();
+            }
         }
+        passFailJFrame.awaitAndCheck();
+    }
+
+    //printable interface
+    public int print(Graphics g, PageFormat pf, int pi) throws PrinterException {
```
Comment (TejeshR13) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:97]: `throws PrinterException` not required.

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 5]
```diff
@@ -24,95 +24,110 @@
...
 
+        if (Files.mismatch(fl.toPath(), fp.toPath()) == -1) {
+            throw new RuntimeException("Printing stream orientation is same for both PORTRAIT and LANDSCAPE");
         }
+    }
 
-        //printable interface
-        public int print(Graphics g, PageFormat pf, int pi) throws PrinterException {
+    //printable interface
+    public int print(Graphics g, PageFormat pf, int pi) throws PrinterException {
```
Comment (TejeshR13) [inline test/jdk/javax/print/StreamPrintingOrientation.java:119]: `throws PrinterException` not required

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java (hunk 3 above)
Comment (prsadhuk) [reply test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:54]: ok

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java (hunk 4 above)
Comment (prsadhuk) [reply test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:97]: it is required...See java.awt.print.Printable spec

File: test/jdk/javax/print/StreamPrintingOrientation.java (hunk 1 above)
Comment (prsadhuk) [reply test/jdk/javax/print/StreamPrintingOrientation.java]: ok

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 6]
```diff
@@ -26,104 +26,120 @@
  * @bug 4869502 4869539
  * @key printer
  * @summary Confirm that ToPage is populated for argument =2. Range is disabled for argument = 0.
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:28]:
```suggestion
 * @summary Confirm that ToPage is populated for argument = 2. Range is disabled for argument = 0.
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 7]
```diff
@@ -26,104 +26,120 @@
...
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.print.PageFormat;
+import java.awt.print.Pageable;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+import java.awt.print.PrinterException;
+
+
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:44]: Extra blank line can be removed.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 8]
```diff
@@ -26,104 +26,120 @@
...
-                System.out.println("print method called "+pi + " Orientation "+pf.getOrientation());
-                return Printable.PAGE_EXISTS;
+        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
+            .title("Instructions")
+            .instructions(INSTRUCTIONS)
+            .rows((int) INSTRUCTIONS.lines().count() + 2)
+            .columns(35)
+            .build();
+
+        System.out.println("open PrintDialog..");
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:81]: print statements can be logged in PFJ's logarea.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 9]
```diff
@@ -26,104 +26,120 @@
...
+            }
         }
+        passFailJFrame.awaitAndCheck();
+    }
+
+    //printable interface
+    public int print(Graphics g, PageFormat pf, int pi) throws PrinterException {
+
+        // Simply draw two rectangles
+        Graphics2D g2 = (Graphics2D)g;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:99]:
```suggestion
        Graphics2D g2 = (Graphics2D) g;
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 10]
```diff
@@ -26,104 +26,120 @@
...
+        g2.setColor(Color.black);
+        g2.translate(pf.getImageableX(), pf.getImageableY());
+        g2.drawRect(1,1,200,300);
+        g2.drawRect(1,1,25,25);
+        System.out.println("print method called "+pi + " Orientation "+pf.getOrientation());
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:104]:
```suggestion
        g2.drawRect(1, 1, 200, 300);
        g2.drawRect(1, 1, 25, 25);
        System.out.println("print method called " + pi + " Orientation " + pf.getOrientation());
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 11]
```diff
@@ -26,104 +26,120 @@
...
-                        pf.setOrientation(PageFormat.PORTRAIT);
-                        System.out.println("Orientation returned from Pageable "+findOrientation(pf.getOrientation()));
-                        return pf;
-                } else {
-                        pf.setOrientation(PageFormat.LANDSCAPE);
-                        System.out.println("Orientation returned from Pageable "+findOrientation(pf.getOrientation()));
-                        return pf;
-                }
+    public PageFormat getPageFormat(int pageIndex) {
+        System.out.println("getPageFormat called "+pageIndex);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:122]:
```suggestion
        System.out.println("getPageFormat called " + pageIndex);
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 12]
```diff
@@ -26,104 +26,120 @@
...
+        if (pageIndex == 0) {
+            pf.setOrientation(PageFormat.PORTRAIT);
+            System.out.println("Orientation returned from Pageable "+findOrientation(pf.getOrientation()));
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:125]:
```suggestion
            System.out.println("Orientation returned from Pageable " + findOrientation(pf.getOrientation()));
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 13]
```diff
@@ -26,104 +26,120 @@
...
+            return pf;
+        } else {
+            pf.setOrientation(PageFormat.LANDSCAPE);
+            System.out.println("Orientation returned from Pageable "+findOrientation(pf.getOrientation()));
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:129]:
```suggestion
            System.out.println("Orientation returned from Pageable " + findOrientation(pf.getOrientation()));
```

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 14]
```diff
@@ -24,95 +24,107 @@
...
+        FileOutputStream fos = null;
+        File fl = null, fp = null;
+        String mType = "application/postscript";
+
+        fl = new File("stream_landscape.ps");
+        fl.deleteOnExit();
+        fos = new FileOutputStream(fl);
+        StreamPrintServiceFactory[] factories = PrinterJob.lookupStreamPrintServices(mType);
+        if (factories.length > 0)
+            service = factories[0].getPrintService(fos);
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:68]:
```suggestion
        if (factories.length > 0) {
            service = factories[0].getPrintService(fos);
        }
```

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 15]
```diff
@@ -24,95 +24,107 @@
...
 
-                try {
-                        f = new File("streamexample.ps");
-                        fos = new FileOutputStream(f);
-                        StreamPrintServiceFactory[] factories = PrinterJob.lookupStreamPrintServices(mType);
-                        if (factories.length > 0)
-                                service = factories[0].getPrintService(fos);
+        System.out.println("\nValues in attr set passed to print method");
+        Attribute attr[] = prSet.toArray();
+        for (int x = 0; x < attr.length; x ++) {
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:85]:
```suggestion
        for (int x = 0; x < attr.length; x++) {
```

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 16]
```diff
@@ -24,95 +24,107 @@
...
+            System.out.println("Name "+attr[x].getName()+"  "+attr[x]);
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:86]:
```suggestion
            System.out.println("Name "+ attr[x].getName() + "  " + attr[x]);
```

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 17]
```diff
@@ -24,95 +24,107 @@
...
+
+        fl = new File("stream_landscape.ps");
+        fl.deleteOnExit();
+        fos = new FileOutputStream(fl);
+        StreamPrintServiceFactory[] factories = PrinterJob.lookupStreamPrintServices(mType);
+        if (factories.length > 0)
+            service = factories[0].getPrintService(fos);
+
+        if (service != null) {
+            System.out.println("Stream Print Service "+service);
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:71]:
```suggestion
            System.out.println("Stream Print Service " + service);
```

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 18]
```diff
@@ -24,95 +24,107 @@
...
-                                throw new RuntimeException("No stream Print Service available.");
-                        }
-                } catch (Exception e) {
-                        e.printStackTrace();
-                }
+        fp = new File("stream_portrait.ps");
+        fp.deleteOnExit();
+        fos = new FileOutputStream(fp);
+        if (factories.length > 0)
+            service = factories[0].getPrintService(fos);
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:99]:
```suggestion
        if (factories.length > 0) {
            service = factories[0].getPrintService(fos);
        }
```

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 19]
```diff
@@ -24,95 +24,107 @@
...
+        pj.setPrintable(pd);
+        prSet.add(OrientationRequested.PORTRAIT);
+        prSet.add(new Copies(1));
+        prSet.add(new JobName("orientation test", null));
+        if (service != null) {
+            pj.print(prSet);
+        }
 
+        if (Files.mismatch(fl.toPath(), fp.toPath()) == -1) {
+            throw new RuntimeException("Printing stream orientation is same for both PORTRAIT and LANDSCAPE");
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:111]: Please limit it to 80 cols.

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 20]
```diff
@@ -24,95 +24,107 @@
...
+        if (pi > 0) {
+            return Printable.NO_SUCH_PAGE;
         }
+        // Simply draw two rectangles
+        Graphics2D g2 = (Graphics2D)g;
+        g2.setColor(Color.black);
+        g2.translate(pf.getImageableX(), pf.getImageableY());
+        System.out.println("StreamPrinting Test Width "+pf.getWidth()+" Height "+pf.getHeight());
+        g2.drawRect(1,1,200,300);
+        g2.drawRect(1,1,25,25);
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:127]:
```suggestion
        System.out.println("StreamPrinting Test Width " + pf.getWidth() + " Height " + pf.getHeight());
        g2.drawRect(1, 1, 200, 300);
        g2.drawRect(1, 1, 25, 25);
```

**Review by kumarabhi006 — COMMENTED** [review]
Minor formatting changes.

**Comment by prsadhuk** [comment]
> Minor formatting changes.

fixed..

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 21]
```diff
@@ -119,14 +116,14 @@ public Printable getPrintable(int pageIndex) {
     }
 
     public PageFormat getPageFormat(int pageIndex) {
-        System.out.println("getPageFormat called "+pageIndex);
+        System.out.println("getPageFormat called " + pageIndex);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:119]: Can be logged by PFJ's logArea.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 22]
```diff
@@ -25,105 +25,118 @@
...
-                                pe.printStackTrace();
-                        }
-                }
+        arg = Integer.parseInt(args[0]);
+
+        String INSTRUCTIONS = " A pagedialog will be shown.";
 
+        if (arg == 0) {
+            INSTRUCTIONS += "\n Confirm that page range is disabled.";
+
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:62]: Blank line not required.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 23]
```diff
@@ -25,105 +25,118 @@
...
-                System.out.println("print method called "+pi + " Orientation "+pf.getOrientation());
-                return Printable.PAGE_EXISTS;
+        if (pj.printDialog()) {
+            try {
+                PassFailJFrame.log("About to print the data ...");
+                pj.print();
+                PassFailJFrame.log("Printed");
+            }
+            catch (PrinterException pe) {
+                pe.printStackTrace();
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:86]:
```suggestion
            } catch (PrinterException pe) {
                pe.printStackTrace();
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 24]
```diff
@@ -25,105 +25,118 @@
...
-                        pf.setOrientation(PageFormat.LANDSCAPE);
-                        System.out.println("Orientation returned from Pageable "+findOrientation(pf.getOrientation()));
-                        return pf;
-                }
+    public PageFormat getPageFormat(int pageIndex) {
+        System.out.println("getPageFormat called " + pageIndex);
+        if (pageIndex == 0) {
+            pf.setOrientation(PageFormat.PORTRAIT);
+            PassFailJFrame.log("Orientation returned from Pageable " + findOrientation(pf.getOrientation()));
+            return pf;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:123]: duplicate code...can be moved outside of `if..else` block.

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java [hunk 25]
```diff
@@ -25,105 +25,118 @@
...
-                } else {
-                        return null;
-                }
+    public String findOrientation(int orient) {
+        if (orient == PageFormat.LANDSCAPE) {
+            return "LANDSCAPE";
+        } else if (orient == PageFormat.PORTRAIT) {
+            return "PORTRAIT";
+        } else if (orient == PageFormat.REVERSE_LANDSCAPE) {
+            return "REVERSE LANDSCAPE";
```
Comment (kumarabhi006) [inline test/jdk/java/awt/print/Dialog/PrintDlgPageable.java:137]: Is it required to check for `REVERSE_LANDSCAPE `as page formatting is set only either `LANDSCAPE `or `PORTRAIT `?

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 26]
```diff
@@ -24,95 +24,110 @@
...
+public class StreamPrintingOrientation implements Printable {
+
+    public static void main(String[] args) throws Exception {
+        StreamPrintingOrientation pd = new StreamPrintingOrientation();
+        PrinterJob pj = PrinterJob.getPrinterJob();
+        HashPrintRequestAttributeSet prSet = new HashPrintRequestAttributeSet();
+        PrintService service = null;
+
+        FileOutputStream fos = null;
+        File fl = null, fp = null;
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:60]: Please declare it on separate lines as per java style coding.

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 27]
```diff
@@ -24,95 +24,110 @@
...
 
-class StreamPrintingOrientation implements Printable {
-        /**
-         * Constructor
-         */
-         public StreamPrintingOrientation() {
-                super();
+        fl = new File("stream_landscape.ps");
+        fl.deleteOnExit();
+        fos = new FileOutputStream(fl);
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:65]: Can be rearranged something like..

    String mType = "application/postscript";
    File fp = null;
    File fl = new File("stream_landscape.ps");
    fl.deleteOnExit();
    FileOutputStream fos = new FileOutputStream(fl);

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 28]
```diff
@@ -24,95 +24,110 @@
...
-                g2.translate(pf.getImageableX(), pf.getImageableY());
-                System.out.println("StreamPrinting Test Width "+pf.getWidth()+" Height "+pf.getHeight());
-                g2.drawRect(1,1,200,300);
-                g2.drawRect(1,1,25,25);
-                return Printable.PAGE_EXISTS;
+        if (pi > 0) {
+            return Printable.NO_SUCH_PAGE;
         }
+        // Simply draw two rectangles
+        Graphics2D g2 = (Graphics2D)g;
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:125]:
```suggestion
        Graphics2D g2 = (Graphics2D) g;
```

File: test/jdk/java/awt/print/Dialog/PrintDlgPageable.java (hunk 24 above)
Comment (prsadhuk) [reply test/jdk/java/awt/print/Dialog/PrintDlgPageable.java]: Removed the whole method as logging for printing is not needed for this PageRange check test

File: test/jdk/javax/print/StreamPrintingOrientation.java [hunk 29]
```diff
@@ -57,10 +57,9 @@ public static void main(String[] args) throws Exception {
         PrintService service = null;
 
         FileOutputStream fos = null;
```
Comment (kumarabhi006) [inline test/jdk/javax/print/StreamPrintingOrientation.java:59]: Can be moved to L64.

**Review by kumarabhi006 — APPROVED** [review]
LGTM except minor change suggested.
